packages/lsp/src/tokens.ts

```diff
@@ -0,0 +1,625 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IDocumentWidget } from '@jupyterlab/docregistry';
+import { ServerConnection } from '@jupyterlab/services';
+import { Token } from '@lumino/coreutils';
+import { IDisposable } from '@lumino/disposable';
+import { ISignal, Signal } from '@lumino/signaling';
+
+import { WidgetLSPAdapter } from './adapters/adapter';
+import { IForeignCodeExtractor } from './extractors/types';
+import { ClientCapabilities, LanguageIdentifier } from './lsp';
+import { LanguageServer1 as LSPLanguageServerSettings } from './plugin';
+import * as SCHEMA from './schema';
+import { VirtualDocument } from './virtual/document';
+import {
+  AnyCompletion,
+  AnyLocation,
+  IDocumentInfo,
+  ILspConnection,
+  ILspOptions
+} from './ws-connection/types';
+
+import type * as rpc from 'vscode-jsonrpc';
+import type * as lsp from 'vscode-languageserver-protocol';
+export { IDocumentInfo };
+export type TLanguageServerId =
```

Comment:
can it be something else? Would it block us if a new server needs to be added?

Reply:
It doesn't matter at runtime, a few example values are used kind of as a sentinel to get typing errors if a different kind of ID (language ID, also a string) was to be mistakenly passed. But no strong feelings on converting it to just a string.

Reply:
Thanks @krassowski happy to keep it that way for now.